### PR TITLE
Disable key-based storage account access in Bicep

### PR DESCRIPTION
**Pull request by pamelafox**
## Purpose

This PR disables key-based access to the storage accounts (both normal and ADLS2) in the Bicep. This doesn't affect standard usage since we use DefaultAzureCredential by default. We do provide the option in prepdocs.py to supply a key, but developers have to come up with that key themselves, so if they really want, they can turn key access back on and use it.

This PR also removes some storage-related lines of seemingly deadcode.

## Does this introduce a breaking change?

When developers merge from main and run the server, azd up, or azd deploy, will this produce an error?
If you're not sure, try it out on an old environment.

```
[ ] Yes
[X] No
```

## Does this require changes to learn.microsoft.com docs?

This repository is referenced by [this tutorial](https://learn.microsoft.com/azure/developer/python/get-started-app-chat-template)
which includes deployment, settings and usage instructions. If text or screenshot need to change in the tutorial,
check the box below and notify the tutorial author. A Microsoft employee can do this for you if you're an external contributor.

```
[ ] Yes
[X] No
```

## Type of change

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[X] Other... Please describe: Security!
```

## Code quality checklist

See [CONTRIBUTING.md](https://github.com/Azure-Samples/azure-search-openai-demo/blob/main/CONTRIBUTING.md#submit-pr) for more details.

- [X] The current tests all pass (`python -m pytest`).
- [ ] I added tests that prove my fix is effective or that my feature works: N/A
- [ ] I ran `python -m pytest --cov` to verify 100% coverage of added lines: N/A
- [x] I ran `python -m mypy` to check for type errors
- [X] I either used the pre-commit hooks or ran `ruff` and `black` manually on my code.

### CODE REVIEW

File: app/backend/approaches/retrievethenread.py [hunk 1]
```diff
@@ -9,10 +8,6 @@
 from core.authentication import AuthenticationHelper
 from core.messagebuilder import MessageBuilder
 
-# Replace these with your own values, either in environment variables or directly here
```
Comment (pamelafox) [inline app/backend/approaches/retrievethenread.py:12]: This was added in the GPT4-vision merge. I'm guessing it was added during development and never removed. We set up the storage accounts in app.py now.

**Comment by mirojs** [comment]
@pamelafox Does it mean that all MS b2c tenant users (both new and old) will have to face more security challenges?

**Comment by pamelafox** [comment]
@mirojs Are you currently using key-based access to storage? This template currently does not use storage keys by default.

**Review by mattgotteiner — APPROVED** [review]
we never used access via key in the backend, and the frontend never directly reaches out to the storage account, so I don't see how this causes any problems

**Comment by mirojs** [comment]
@pamelafox No, I'm not. But all the b2c auth doesn't support RBAC, which leaves a vulnerable spot there in the app. Is my understanding correct?

**Comment by pamelafox** [comment]
@mirojs Ah, you are correct that currently we use client secret when authenticating our Entra app registrations with the MSAL SDK. We would like to move away from that secret and are working with the Entra teams to figure out how we can do that. Stay tuned!

**Pull request merged**